### PR TITLE
Create an every element SmartApp equivalent for Device Tiles

### PR DESCRIPTION
**Pull request by KyleLeneau**
- [x] Service Manager like SmartApp to add/remove sample DTH's
- [x] Tile Multiattribute Generic
- [x] Tile Multiattribute Lighting
- [x] Tile Multiattribute Thermostat
- [x] Tile Multiattribute Media Player
- [x] Tile Multiattribute Video Player
- [x] Tile Device Standard
- [x] Tile Device Value
- [ ] Tile Device Presence
- [x] Tile Device Carousel
- [x] Tile Device Slider
- [x] Tile Device Color Wheel

### CODE REVIEW

**Comment by StrykerSKS** [comment]
These examples seem to be using the depricated thermostat VALUE_CONTROL model:

> tileAttribute("device.temperature", key: "VALUE_CONTROL") {
> -              attributeState("default", action: "setTemperature")
> -          }

http://docs.smartthings.com/en/latest/device-type-developers-guide/tiles-metadata.html#a-word-on-the-value-control-attribute

Instead of "default", "VALUE_UP" and "VALUE_DOWN" should be used.

**Comment by KyleLeneau** [comment]
These aren't deprecated per say, these are options on a `VALUE_CONTROL` and some DTH's still use them so the mobile apps still need to support them. Although like you said they are now the recommended way.

**Comment by StrykerSKS** [comment]
Well, using "default" is broken and sends inconsistent values. So unless that is planned to be fixed or documented then I would not recommend using it.

**Comment by KyleLeneau** [comment]
Check the latest commit, we just merged something in for the update.

**Comment by juliedolanjohnson** [comment]
Hey guys - tried to install the the thermostat DTH here and had to add a comma after "thermostatBasic" in the main tile details, could that get updated?

**Comment by pennig** [comment]
Carousel is in there for review, but it's not working. Not sure why. @KyleLeneau, can we get a DTH person who knows how cameras work to look over it?

File: devicetypes/smartthings/tile-ux/tile-basic-carousel.src/tile-basic-carousel.groovy [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+def fanAuto() {
+	sendEvent(name: "thermostatFanMode", value: "fanAuto")
+}
+
+def fanCirculate() {
+	sendEvent(name: "thermostatFanMode", value: "fanCirculate")
+}
+
+def poll() {
+	null
```
Comment (workingmonk) [inline devicetypes/smartthings/tile-ux/tile-basic-carousel.src/tile-basic-carousel.groovy]: if you don't need the poll method, you can remove it

File: devicetypes/smartthings/tile-ux/tile-basic-presence.src/tile-basic-presence.groovy [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+ *
+ */
+metadata {
+	definition (
+		name: "presenceDeviceTile",
+		namespace: "smartthings/tile-ux",
+		author: "SmartThings") {
+
+		capability "Thermostat"
+		capability "Relative Humidity Measurement"
```
Comment (workingmonk) [inline devicetypes/smartthings/tile-ux/tile-basic-presence.src/tile-basic-presence.groovy]: why does the presence tile have thermostat code?

File: devicetypes/smartthings/tile-ux/tile-basic-standard.src/tile-basic-standard.groovy [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	}
+}
+
+def installed() {
+	sendEvent(name: "switch", value: "off")
+}
+
+def parse(String description) {
+	def pair = description.split(":")
+	createEvent(name: pair[0].trim(), value: pair[1].trim())
```
Comment (workingmonk) [inline devicetypes/smartthings/tile-ux/tile-basic-standard.src/tile-basic-standard.groovy]: what are you expecting the parse method to get here? "switch: on" ??

File: devicetypes/smartthings/tile-ux/tile-basic-value.src/tile-basic-value.groovy [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+	sendEvent(name: "text", value: "Test")
+	sendEvent(name: "longText", value: "The Longer The Text, The Better The Test")
+	sendEvent(name: "integer", value: 47)
+	sendEvent(name: "integerFloat", value: 47.0)
+	sendEvent(name: "pi", value: 3.14159)
+	sendEvent(name: "floatAsText", value: "3.14159")
+}
+
+def parse(String description) {
+}
```
Comment (workingmonk) [inline devicetypes/smartthings/tile-ux/tile-basic-value.src/tile-basic-value.groovy:96]: might help to put a new line here

File: devicetypes/smartthings/tile-ux/tile-multiattribute-generic.src/tile-multiattribute-generic.groovy [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	if (level < 100) {
+		level = level + 1
+	}
+	setLevel(level)
+}
+
+def levelDown() {
+	def level = device.latestValue("level") as Integer ?: 0
+	if (level > 0) {
+		level = level - 1
```
Comment (workingmonk) [inline devicetypes/smartthings/tile-ux/tile-multiattribute-generic.src/tile-multiattribute-generic.groovy:115]: you are dimming this 1 point at a time?

**Comment by workingmonk** [comment]
completed a quick review and added the comments

File: devicetypes/smartthings/tile-ux/tile-multiattribute-generic.src/tile-multiattribute-generic.groovy (hunk 5 above)
Comment (KyleLeneau) [inline devicetypes/smartthings/tile-ux/tile-multiattribute-generic.src/tile-multiattribute-generic.groovy:115]: Yeah, it's for the up and down arrows for the value control.

**Comment by KyleLeneau** [comment]
@workingmonk Back to you, I addressed all the issues and squashed all the changes to one commit.

File: devicetypes/smartthings/tile-ux/tile-multiattribute-generic.src/tile-multiattribute-generic.groovy [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+		}
+		multiAttributeTile(name:"sliderTile", type:"generic", width:6, height:4) {
+			tileAttribute("device.switch", key: "PRIMARY_CONTROL") {
+				attributeState "on", label:'${name}', backgroundColor:"#79b821", nextState:"turningOff"
+				attributeState "off", label:'${name}', backgroundColor:"#ffffff", nextState:"turningOn"
+				attributeState "turningOn", label:'${name}', backgroundColor:"#79b821", nextState:"turningOff"
+				attributeState "turningOff", label:'${name}', backgroundColor:"#ffffff", nextState:"turningOn"
+			}
+			tileAttribute("device.level", key: "SECONDARY_CONTROL") {
+				attributeState "default", icon: 'st.Weather.weather1', action:"randomizeLevel"
```
Comment (adammika) [inline devicetypes/smartthings/tile-ux/tile-multiattribute-generic.src/tile-multiattribute-generic.groovy:42]: This icon appears very small (due to the large amounts of whitespace surrounding the image) and non-white. We should probably have another icon here.

**Comment by workingmonk** [comment]
i am good with the quick review. you addressing the comments and squashing them :+1:

**Pull request merged**